### PR TITLE
Combines map instance members: `once`, `off` (to match `on`)

### PR DESCRIPTION
**Pull request by domlet**
Closes: https://github.com/mapbox/mapbox-gl-js-docs/issues/340

- Extends the pattern of collapsing multiple versions of instance members implemented in a previous docs sprint
- Changes `off` and `once` to appear once each instead of twice
- Uses complete sentences for single-line code comments
- Adds examples for `once` and `off`
- Adds links to related examples for `on`, `once`, and `off`

## screenshots

before/after:
 screenshot --> screenshot

before/after:
screenshot --> screenshot 

## Questions for review
1. Does the inline example for `off` make sense?
2. How can I add the `?` to appear after the optional argument (`layerId`) like it does for [`panBy`](http://localhost:65257/mapbox-gl-js/api/map/#map#panby)?

Tagging @HeyStenson @mapbox/gl-js for review.

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [x] include before/after visuals or gifs if this PR includes visual changes
 - [x] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'

### CODE REVIEW

**Review by ryanhamley — CHANGES_REQUESTED** [review]
This makes sense and the `off` example is good but changing the `layerId` to be optional leads to Flow errors because Flow now thinks that the variable passed to `Evented.on/off` as a listener might be undefined. We know that's not the case, but Flow isn't that smart. I think it makes sense though because technically speaking, it's the third argument that's optional, not the second one. But it's also not correct really to list the listener as optional. It's just kind of an odd pattern to fit into Flow.

What you can do to get around this is to change the `if (listener === undefined) {` line to `if (listener === undefined && typeof layerId === 'function') {` which refines `layerId` and tells Flow that we're definitely passing a function to `Evented.on/off`.

As for why the `?` isn't showing up, my guess is that it's either because Flow had an error or because the optional argument is followed by a non-optional argument. This may be an error with the code that builds the docs that would need to be corrected but I'm not super familiar with that.

**Review by HeyStenson — APPROVED** [review]
@domlet this is a 👍  from me in terms of WORDS. Lmk if you want to tackle the `?` issue together.

**Comment by domlet** [comment]
Thank you @ryanhamley 

1. I'll continue investigating the `?` issue in https://github.com/mapbox/mapbox-gl-js-docs/issues/592
2. I implemented the Flow workaround in 90a9b77 – but with or without those changes `test-flow` fails to run with `npm test`:

```
➜  mapbox-gl-js git:(db-mapcombineinstancemembers) ✗ npm test

> mapbox-gl@2.4.0-dev test
> run-s lint lint-css lint-docs test-flow test-unit

> mapbox-gl@2.4.0-dev lint
> eslint --cache --ignore-path .gitignore src test bench debug/*.html
...
> mapbox-gl@2.4.0-dev lint-css
> stylelint 'src/css/mapbox-gl.css'

> mapbox-gl@2.4.0-dev lint-docs
> documentation lint src/index.js

> mapbox-gl@2.4.0-dev test-flow
> build/run-node build/generate-flow-typed-style-spec && flow .

node: bad option: --experimental-loader
ERROR: "test-flow" exited with 9.```
```

Any advice on passing these tests so I can merge this update to the API docs? Thank you!

File: src/ui/map.js [hunk 1]
```diff
@@ -1117,36 +1118,29 @@ class Map extends Camera {
...
+     * optionally limited to events occurring on features in a specified style layer.
      *
      * @param {string} type The event type to listen for; one of `'mousedown'`, `'mouseup'`, `'click'`, `'dblclick'`,
      * `'mousemove'`, `'mouseenter'`, `'mouseleave'`, `'mouseover'`, `'mouseout'`, `'contextmenu'`, `'touchstart'`,
      * `'touchend'`, or `'touchcancel'`. `mouseenter` and `mouseover` events are triggered when the cursor enters
      * a visible portion of the specified layer from outside that layer or outside the map canvas. `mouseleave`
      * and `mouseout` events are triggered when the cursor leaves a visible portion of the specified layer, or leaves
      * the map canvas.
-     * @param {string} layerId The ID of a style layer. Only events whose location is within a visible
+     * @param {string} layerId (optional) The ID of a style layer. Only events whose location is within a visible
```
Comment (ryanhamley) [inline src/ui/map.js:1130]: Maybe it's worth expanding the descriptions of `layerId` to note that when it's not provided, the listener will be triggered by the corresponding event happening anywhere on the map.

File: src/ui/map.js [hunk 2]
```diff
@@ -1117,36 +1118,29 @@ class Map extends Camera {
...
      * feature in this layer will trigger the listener. The event will have a `features` property containing
      * an array of the matching features.
      * @param {Function} listener The function to be called when the event is fired.
      * @returns {Map} `this`
+     * @example
+     * // Log the coordinates of a user's first map touch.
+     * map.once('touchstart', function (e) {
```
Comment (ryanhamley) [inline src/ui/map.js:1137]: Should we include an example with the layerId as well? Or do you think that's redundant?

**Review by ryanhamley — APPROVED** [review]
Hmm, I don't see any problems when I check out that commit and run it locally. I'm not sure why you were having issues.

I think what you have here is a good compromise though. It's kind of annoying to have to add things to the code strictly for Flow and this approach sidesteps the issue of the `?` not appearing in the docs. I just had a couple thoughts but it looks good.

File: src/ui/map.js (hunk 2 above)
Comment (domlet) [reply src/ui/map.js:1137]: Good catch – added in 3b949dfb8.

File: src/ui/map.js (hunk 1 above)
Comment (domlet) [reply src/ui/map.js]: That's super helpful. Added in b00322750 🏆

File: src/ui/map.js (hunk 2 above)
Comment (domlet) [reply src/ui/map.js:1142]: Simplified the example in fbad5875b.

**Comment by domlet** [comment]
Thanks for all your help @ryanhamley 👍🏼 

I'll continue investigating the `?` issue in mapbox/mapbox-gl-js-docs#592 because when an argument is optional it should appear with the `?` annotation.

**Review by ryanhamley — APPROVED** [review]
Looks great!

**Pull request merged**